### ISSUE TITLE
linux-raspberrypi can not be built with pmbootstrap

### ISSUE THREAD

**Issue by ollieparanoid** [postmarketOS/pmbootstrap#1514]
Reported by @yangxuan8282 in https://github.com/postmarketOS/pmbootstrap/issues/1507#issuecomment-391219470:

```
>>> linux-raspberrypi: Applying rpi-4.14.39-alpine.patch...
>>> linux-raspberrypi: Applying issue-4973.patch...
make: Entering directory '/home/pmos/build/src/linux-4.14'
make[1]: Entering directory '/home/pmos/build/src/build-raspberrypi'
getconf: LFS_CFLAGS: unknown variable
getconf: LFS_LDFLAGS: unknown variable
getconf: LFS_LIBS: unknown variable
  HOSTCC  scripts/basic/fixdep
  GEN     ./Makefile
/home/pmos/build/src/linux-4.14/scripts/basic/fixdep.c:105:23: fatal error: sys/types.h: No such file or directory
 #include <sys/types.h>
                       ^
compilation terminated.
make[2]: *** [scripts/Makefile.host:102: scripts/basic/fixdep] Error 1
make[1]: *** [/home/pmos/build/src/linux-4.14/Makefile:468: scripts_basic] Error 2
make[1]: Leaving directory '/home/pmos/build/src/build-raspberrypi'
make: *** [Makefile:146: sub-make] Error 2
make: Leaving directory '/home/pmos/build/src/linux-4.14'
>>> ERROR: linux-raspberrypi: all failed
```

(The package was a bit modified, but he said the original `linux-raspberrypi` from Alpine caused the same error. Adding `linux-headers` to makedepends didn't help.)

**Comment by yangxuan8282** [comment]
before find out what caused this issues, I guess even create a linux-rpi PR can't merge, because for now it can't compile with pmbootstrap

**Comment by ollieparanoid** [comment]
True. Unfortunately I could not find out *why* this happens. I guess we can compare it with the regular `linux-vanilla` APKBUILD and see what it does differently, I'll try to get that done next week.

**Comment by yangxuan8282** [comment]
seems compare to `linux-vanilla`, the `linux-rpi` not include cross compile stuff
in `linux-rpi` APKBUILD it is:

```
make ...
```

on vanilla it's: 

```
make ARCH="$_carch" HOSTCC="$HOSTCC" ...
```

**Comment by ollieparanoid** [comment]
Great finding! I've tried to add that variable, but then it fails with:
> `/home/pmos/build/src/linux-4.9/Makefile:534: arch/armhf/Makefile: No such file or directory`

So we need to adjust it some more to make it work. Maybe it's easier to start with the current linux-vanilla APKBUILD from Alpine and apply the Pi specific changes on top of that?

**Comment by yangxuan8282** [comment]
@ollieparanoid 
that's because there is no arch called armhf in linux arch folder, it's arm

linux-vanilla convert it in this way:

```
_carch=${CARCH}
case "$_carch" in
aarch64*) _carch="arm64" ;;
arm*) _carch="arm" ;;
mips*) _carch="mips" ;;
ppc*) _carch="powerpc" ;;
s390*) _carch="s390" ;;
esac
```

but seems still need some other changes

**Comment by yangxuan8282** [comment]
@ollieparanoid 
I have update my linux-rpi APKBUILD, add cross compile, so it can be build with pmbootstrap now

https://github.com/yangxuan8282/rpi-aports/blob/master/linux-rpi/APKBUILD

**Comment by yangxuan8282** [comment]
when the build come to tracing dependencies, got this issues:

```
...
>>> linux-rpi2-dev*: Preparing subpackage linux-rpi2-dev...
>>> linux-rpi2-dev*: Running postcheck for linux-rpi2-dev
>>> linux-rpi*: Running postcheck for linux-rpi
>>> linux-rpi*: Preparing package linux-rpi...
>>> linux-rpi-dev*: Scanning shared objects
>>> linux-rpi*: Scanning shared objects
>>> linux-rpi2-dev*: Scanning shared objects
>>> linux-rpi2*: Scanning shared objects
>>> linux-rpi-dev*: Tracing dependencies...
ERROR: Unable to open root: No such file or directory
ERROR: Failed to open apk database: No such file or directory
ERROR: Unable to open root: No such file or directory
ERROR: Failed to open apk database: No such file or directory
>>> ERROR: linux-rpi2*: create_apks failed
>>> ERROR: linux-rpi: all failed
```

have to add `!tracedeps` in linux-rpi options section

**Comment by ollieparanoid** [comment]
Yeah, since we compile kernels in the "native" cross compiling mode (everything runs in the native chroot, distcc is not involved), "!tracedeps" needs to be enabled:

https://github.com/postmarketOS/pmbootstrap/blob/6cffe6d0589c24ac0147bc378aa884285df9b034/pmb/build/_package.py#L328-L331


Heads up regarding the `-dev` subpackage, in case you run into further errors with it. It is special, because it will package the little "scripts" (natively compiled programs) that are required to  build the kernel. In the "native" cross compiling mode, these get compiled for the native architecture (`x86_64`), so the `-dev` package isn't the target architecture but the native architecture instead.

The `linux-vanilla` package supports cross compiling with setting environment variables (we use that internally), and if one is cross compiling, it would change the architecture of the `-dev` package to be of the host architecture. We made a workaround for that in #1234.

If you don't need the `-dev` subpackage and the build runs through with `!tradeceps` however, I recommend to leave the dev subpackaging of `linux-raspberrypi` as it is right now.

**Comment by yangxuan8282** [comment]
So, can I create a linux-rpi PR use current APKBUILD?

**Comment by yangxuan8282** [comment]
I guess we can close this issues now
